packages/model-viewer/src/features/scene-graph/image.ts

```diff
@@ -20,34 +20,34 @@ import {Image as GLTFImage} from '../../three-components/gltf-instance/gltf-2.0.
 import {Renderer} from '../../three-components/Renderer.js';
 
 import {Image as ImageInterface} from './api.js';
-import {$gltfImage, $threeTexture, TextureInfo} from './texture-info.js';
-import {$onUpdate, $sourceObject, ThreeDOMElement} from './three-dom-element.js';
+import {$correlatedObjects, $onUpdate, $sourceObject, ThreeDOMElement} from './three-dom-element.js';
 
 
 
 const loader = new ImageLoader();
 const quadMaterial = new MeshBasicMaterial();
 const quad = new PlaneGeometry(2, 2);
 
-export const $underlyingTexture = Symbol('threeTextures');
-const $textureInfo = Symbol('textureInfo');
+export const $sourceTexture = Symbol('sourceTexture');
 export const $applyTexture = Symbol('applyTexture');
 
 /**
  * Image facade implementation for Three.js textures
  */
 export class Image extends ThreeDOMElement implements ImageInterface {
-  get[$underlyingTexture]() {
-    return this[$textureInfo][$threeTexture];
+  get[$sourceTexture]() {
+    console.assert(
+        this[$correlatedObjects] != null && this[$correlatedObjects]!.size > 0,
+        'Sampler correlated object is undefined');
+    return this[$correlatedObjects]?.values().next().value as ThreeTexture;
   }
 
-  private[$textureInfo]: TextureInfo;
-  constructor(textureInfo: TextureInfo) {
-    super(
-        textureInfo.onUpdate,
-        textureInfo[$gltfImage],
-        new Set<ThreeTexture>([textureInfo[$threeTexture]!]));
-    this[$textureInfo] = textureInfo;
+  constructor(
+      onUpdate: () => void, texture: ThreeTexture|null,
+      gltfImage: GLTFImage|null) {
+    gltfImage =
+        gltfImage ?? {name: 'adhoc_image', uri: 'adhoc_image'} as GLTFImage;
```

Comment:
If we have a ThreeTexture, let's set the uri here to its URI.

Reply:
Done.